### PR TITLE
add option to disable remote ca verify

### PR DESCRIPTION
**Pull request by Shikugawa**
This is needed for #201.

### CODE REVIEW

File: test/filters/oidc/oidc_filter_test.cc [hunk 1]
```diff
@@ -1128,7 +1138,9 @@ google::rpc::Code OidcFilterTest::MakeRequestWhichWillCauseTokenRetrieval(
   auto raw_http = common::http::response_t(
       new beast::http::response<beast::http::string_body>());
   raw_http->result(beast::http::status::ok);
-  EXPECT_CALL(*mocked_http, Post(Eq(token_uri), _, _, Eq("some-ca"),
+  const common::http::TransportSocketOptions opt{.ca_cert_ = "some-ca",
```
Comment (incfly) [inline test/filters/oidc/oidc_filter_test.cc:1141]: will it make sense to have a `kDefaultTransportSocketOpts` constant defined in the beginning?

File: src/common/http/http.h [hunk 2]
```diff
@@ -45,6 +47,17 @@ class PathQueryFragment {
   inline bool HasFragment() const { return !fragment_.empty(); }
 };
 
+struct TransportSocketOptions {
+  // the ca cert to be trusted in the http call
```
Comment (incfly) [inline src/common/http/http.h:51]:
```suggest
  // the ca cert to be trusted in the http call, in PEM format.
```

Comment (incfly) [reply src/common/http/http.h:51]: If I understand correctly, it's inlined PEM format.

**Review by incfly — COMMENTED** [review]
LGTM, just a few nitpicks.

**Comment by bruegth** [comment]
@incfly 
Did you forgot this RFC thing (https://github.com/istio-ecosystem/authservice/pull/126#issuecomment-934655392) ?

But nice to have this now. 👍🏻

**Comment by Shikugawa** [comment]
> @incfly Did you forgot this RFC thing ([#126 (comment)](https://github.com/istio-ecosystem/authservice/pull/126#issuecomment-934655392)) ?
> 
> But nice to have this now. 👍🏻

We have made it clear to the API that this feature should only be used in a test environment. (This is required for integration testing with self-signed certificates.)

**Comment by bruegth** [comment]
@Shikugawa Sure, but result would be the same or? 

And if an ODICP instance is behind an Reverse Proxy (TLS endpoint) on same host, it still makes no sense to put an certificate onto it.

**Comment by Shikugawa** [comment]
@bruegth The IDP setting of authservice is implemented to accept only HTTPS, and adding an option to allow HTTP to it is not desirable. Therefore, when configuring an external IDP for integration testing, a TLS certificate should always be set. However, it is practical to use self-signed certificates in a test environment, and for that reason, we should be able to disable certificate validation.

**Comment by incfly** [comment]
When reviewing PR, I actually did worry about this can cause misunderstanding like this, since it's defined as part of the API proto. @Shikugawa for integration test we could provide a way to specify additional root (even if it's self signed). Istio does the similiar things for its JWKS endpoint HTTPS fetching.

**Comment by Shikugawa** [comment]
I also had concerns about providing this feature as an API, so I considered several other ways (e.g., providing it as a runtime option, or switching it in a compile option), but I thought that an API would be fine as long as it is clearly documented. However, if @incfly has other concerns, we can provide a better way. We can just not include this change in the next release.

**Pull request merged**